src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -0,0 +1,1403 @@
+}
+
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<real,nstate> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convective_dissipative_source_term (
+        const dealii::Point<dim,real> &pos,
+        const std::array<real,nstate> &/*solution*/,
+        const dealii::types::global_dof_index cell_index) const
+{
+    std::array<real,nstate> conv_source_term = convective_source_term(pos);
+    std::array<real,nstate> diss_source_term = dissipative_source_term(pos,cell_index);
+    std::array<real,nstate> convective_dissipative_source_term;
+    for (int s=0; s<nstate; s++)
+    {
+        convective_dissipative_source_term[s] = conv_source_term[s] + diss_source_term[s];
+    }
+    return convective_dissipative_source_term;
+}
+
+// Returns the value from a CoDiPack or Sacado variable.
+template<typename real>
+double getValue(const real &x) {
+    if constexpr(std::is_same<real,double>::value) {
+        return x;
+    }
+    else if constexpr(std::is_same<real,FadType>::value) {
+        return x.val(); // sacado
+    } 
+    else if constexpr(std::is_same<real,FadFadType>::value) {
+        return x.val().val(); // sacado
+    }
+    else if constexpr(std::is_same<real,RadType>::value) {
+      return x.value(); // CoDiPack
+    } 
+    else if(std::is_same<real,RadFadType>::value) {
+        return x.value().value(); // CoDiPack
+    }
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+dealii::Tensor<2,nstate,real> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convective_flux_directional_jacobian (
+    const std::array<real,nstate> &conservative_soln,
+    const dealii::Tensor<1,dim,real> &normal) const
+{
+    using adtype = FadType;
+
+    // Initialize AD objects
+    std::array<adtype,nstate> AD_conservative_soln;
+    for (int s=0; s<nstate; s++) {
+        adtype ADvar(nstate, s, getValue<real>(conservative_soln[s])); // create AD variable
+        AD_conservative_soln[s] = ADvar;
+    }
+
+    // Compute AD convective flux
+    std::array<dealii::Tensor<1,dim,adtype>,nstate> AD_convective_flux = convective_flux_templated<adtype>(AD_conservative_soln);
+
+    // Assemble the directional Jacobian
+    dealii::Tensor<2,nstate,real> jacobian;
+    for (int sp=0; sp<nstate; sp++) {
+        // for each perturbed state (sp) variable
+        for (int s=0; s<nstate; s++) {
+            jacobian[s][sp] = 0.0;
+            for (int d=0;d<dim;d++) {
+                // Compute directional jacobian
+                jacobian[s][sp] += AD_convective_flux[s][d].dx(sp)*normal[d];
+            }
+        }
+    }
+    return jacobian;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+dealii::Tensor<2,nstate,real> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::dissipative_flux_directional_jacobian (
```

Comment:
This and a few other functions are being repeated in NavierStokes, RANS and LargeEddySimulationBase. Maybe these common functions can be moved to a separate class which NavierStokes, RANS and LargeEddySimulationBase can access.

Reply:
Maybe I can do this in the later PR

Reply:
Creating something like `TurbulenceModellingBase`? Open an issue for this

Reply:
Can resolve this once an issue has actually been opened